src/Exclusion/ExpressionLanguageExclusionStrategy.php

```diff
@@ -56,4 +57,31 @@ public function shouldSkipProperty(PropertyMetadata $property, Context $navigato
 
         return $this->expressionEvaluator->evaluate($property->excludeIf, $variables);
     }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    public function shouldSkipClass(ClassMetadata $class, Context $navigatorContext): bool
+    {
+        if (null === $class->excludeIf) {
+            return false;
+        }
+
+        $variables = [
+            'context' => $navigatorContext,
+            'property_metadata' => $class,
+        ];
+        if ($navigatorContext instanceof SerializationContext) {
+            $variables['object'] = $navigatorContext->getObject();
+        } else {
```

Comment:
I think you can avoid using `else` by adding default value in `$variables` array:

```diff
        $variables = [
            'context' => $navigatorContext,
            'property_metadata' => $class,
+           'object' => null,
        ];

        if ($navigatorContext instanceof SerializationContext) {
            $variables['object'] = $navigatorContext->getObject();
-       } else {
-           $variables['object'] = null;
        }
```